resources/js/views/global/Supercluster.vue

```diff
@@ -184,177 +183,23 @@ function onEachArtFeature (feature, layer)
             duration: 10
         });
 
-        const user = (feature.properties.name || feature.properties.username)
-            ? `By ${feature.properties.name ? feature.properties.name : ''} ${ feature.properties.username ? '@' + feature.properties.username : ''}`
-            : "";
+        const user = mapHelper.formatUserName(feature.properties.name, feature.properties.username);
 
-        const team = (feature.properties.team)
-            ? `\nTeam ${feature.properties.team}`
-            : "";
-
-        // todo - increase the dimensions of each art image
-        L.popup()
+        L.popup(mapHelper.popupOptions)
             .setLatLng(feature.geometry.coordinates)
             .setContent(
-                '<img src= "' + feature.properties.filename + '" class="litter-img art-litter-image" />'
-                    + '<div class="litter-img-container">'
-                        + '<p>Taken on ' + moment(feature.properties.datetime).format('LLL') +'</p>'
-                        + user
-                        + team
-                    + '</div>'
+                mapHelper.getMapImagePopupContent(
+                    feature.properties.filename,
+                    null,
+                    feature.properties.datetime,
+                    user,
+                    feature.properties.team
+                )
             )
             .openOn(map);
     });
 }
 
-/**
- * The user dragged or zoomed the map, or changed a category
- */
-async function update ()
```

Comment:
We want to separate some Vue and Map state. This update was called by change map. Does it still work moved back into vue methods?

Reply:
I actually wanted to move some logic into Vue instance methods to possibly use Vue reactivity to optimize things a bit. However, I've only almost copy-pasted the method without changing anything, and I've tested that it works just as it did before.

I think that the code for the maps still needs a lot of refactoring, and moving code from functions into objects is a good step in the right direction. Since the code for the Global map is different from the one for Profile, Teams, and Cities I didn't change it, just moved a part of it into Vue. What do you think?

Reply:
Using more vue would be great but I found it tricky to combine leaflet and vue. If we have to keep them separate, we could create another mapFunctionHelper